api/src/main/java/org/wildfly/transformer/TransformerBuilder.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.transformer;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ConcurrentModificationException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static java.lang.Thread.currentThread;
+
+/**
+ * Resource transformer builder. Instances of this class are thread safe.
```

Comment:
Are instances thread safe (multiple threads can safely use an instance) or thread specific (only usable from same thread that instances was created)?  

The thread check in setPackagesMapping() looks to be thread specific.

Reply:
Also, did you consider using ThreadLocal?  ThreadLocal is not perfect and can be the source of memory leaks, but could be an alternative.

Reply:
I updated the javadoc accordingly @scottmarlow